CMakeLists.txt

```diff
@@ -317,6 +317,19 @@ endif()
 zephyr_compile_options($<$<COMPILE_LANGUAGE:C>:$<TARGET_PROPERTY:compiler,warning_base>>)
 zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:$<TARGET_PROPERTY:compiler-cpp,warning_base>>)
 
+# @Intent: Set XTENSA_TOOLCHAIN_VARIANT macro required for Xtensa coredump
```

Comment:
does this really belong here in the top cmake file? Can you maybe detect toolchain at runtime by examining toolchain macros where this code would end up being used? instead of doing it this way? @stephanosio  your input please and how we could detect the zephyr toolchain for example...

Reply:
> your input please and how we could detect the zephyr toolchain for example

I am afraid there currently is no way to detect which "flavour" of GCC (XCC, Zephyr SDK, Espressif, ...) you are using in the code.

IMHO, a more generic solution would be preferred if we are to do this; for example:
```
string(TOUPPER ${ZEPHYR_TOOLCHAIN_VARIANT} ZEPHYR_TOOLCHAIN_VARIANT_UPPERCASE)
zephyr_compile_definitions(__TOOLCHAIN_${ZEPHYR_TOOLCHAIN_VARIANT_UPPERCASE}__)
```

cc @tejlmand